### PR TITLE
449 refacto udviz multi packages

### PR DESCRIPTION
**Pull request by valentinMachado**

### CODE REVIEW

File: packages/core/src/Game/GameObject/GameObject.js [hunk 1]
```diff
@@ -168,40 +169,38 @@ const GameObjectModule = class GameObject {
...
+
+  isInitialized() {
+    return this.initialized;
   }
 
   /**
    * Return the world transform of this
    *
-   * @returns {Transform}
+   * @returns {object}
```
Comment (valentinMachado) [inline packages/core/src/Game/GameObject/GameObject.js:184]: when object in jsdoc precise object fields toJSON should do it

File: packages/core/src/Game/GameObject/GameObject.js [hunk 2]
```diff
@@ -251,13 +250,13 @@ const GameObjectModule = class GameObject {
    * @returns {object}
    */
   fetchWorldScripts() {
```
Comment (valentinMachado) [inline packages/core/src/Game/GameObject/GameObject.js:252]: should be a method of context (world or browser)

File: packages/core/src/Game/GameObject/GameObject.js [hunk 3]
```diff
@@ -874,7 +795,7 @@ GameObjectModule.TYPE = 'GameObject';
  * @param {number} ratio a number between 0 => 1
  * @returns {GameObject} g1 interpolated
  */
-GameObjectModule.interpolateInPlace = function (g1, g2, ratio) {
+GameObject.interpolateInPlace = function (g1, g2, ratio) {
```
Comment (valentinMachado) [inline packages/core/src/Game/GameObject/GameObject.js:798]: interpolate should interpolate browserscript/worldscript ? conf numeric values ?

File: packages/core/src/Game/World.js [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+ * Represent a world model and methods to simulated it given a {@link WorldContext}
+ */
+const World = class {
+  /**
+   *
+   * @param {object|JSON} json - The json object that contains the world's data.
+   * @param {object|JSON} json.gameObject - JSON to create the {@link GameObject} of the world.
+   * @param {string} json.name - Name of the world. (optional)
+   * @param {string} json.uuid - UUID of the world. (optional)
+   * @param {WorldOrigin} json.origin - Origin of the world. (optional)
```
Comment (valentinMachado) [inline packages/core/src/Game/World.js:28]: precise you need pass it in order to initialize the itownsView ? (we are in core there are no notion of itowns here)

File: packages/core/src/Game/World.js [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+    }
+  }
+
+  /**
+   * Load its gameobject
+   *
+   * @param {Function} onLoad callback called at the end of the load
+   * @param {WorldContext} worldContext world context to initialize the gameobject
+   */
+  load(onLoad, worldContext) {
```
Comment (valentinMachado) [inline packages/core/src/Game/World.js:92]: replace onLoad by a Promise

File: packages/core/src/Game/World.js [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+   * @param {GameObject} go the gameobject to compute load promises
+   * @returns {Array[Promise]} An array containing all the promises
+   */
+  computePromisesLoad(go) {
+    const promises = [];
+
+    go.traverse(function (g) {
+      const scriptC = g.getComponent(WorldScript.Model.TYPE);
+      if (scriptC) {
+        const scripts = scriptC.getController().getScripts();
```
Comment (valentinMachado) [inline packages/core/src/Game/World.js:108]: gameview should do the same with onComponentUpdate execute script by script

File: packages/core/src/Game/World.js [hunk 7]
```diff
@@ -0,0 +1,429 @@
...
+      );
+      if (worldScriptGameComponent) {
+        worldScriptGameComponent.getController().execute(event, params);
+      }
+    });
+  }
+
+  /**
+   * Add a GameObject into this world
+   * Then call a callback onLoad (optional)
```
Comment (valentinMachado) [inline packages/core/src/Game/World.js:145]: remove onLoad with a Promise

File: packages/core/src/Game/World.js [hunk 8]
```diff
@@ -0,0 +1,429 @@
...
+   *
+   * @param {GameObject} gameObject - The gameobject to add
+   * @param {WorldContext} worldContext - The world context to initialize the gameobject
+   * @param {GameObject} parent - The gameobject parent may be null
```
Comment (valentinMachado) [inline packages/core/src/Game/World.js:149]: precise optional and behavior associated

File: packages/core/src/Game/World.js [hunk 9]
```diff
@@ -0,0 +1,429 @@
...
+   * @param {Function} onLoad - Callback called when loaded (optional)
+   */
+  addGameObject(gameObject, worldContext, parent, onLoad = null) {
+    const _this = this;
+
+    worldContext
```
Comment (valentinMachado) [inline packages/core/src/Game/World.js:155]: remove param worldcontext and do the init in worldcontext class

File: packages/core/src/Game/World.js [hunk 10]
```diff
@@ -0,0 +1,429 @@
...
+            );
+            buffer.splice(i, 1); // Remove from buffer
+          }
+        }
+      }
+    });
+  }
+
+  /**
+   * @param {boolean} withWorldComponent - If true, the world component will be included in the world state
```
Comment (valentinMachado) [inline packages/core/src/Game/World.js:357]: precise isOutdated is reset here

**Comment by valentinMachado** [comment]
gameobject should have a method editConfBrowserScript which write the new conf and setOutdated to true

**Comment by valentinMachado** [comment]
in order to write jsdoc of interface of the baseScript (world + browser) they should be write manually

File: packages/core/src/Game/World.js (hunk 9 above)
Comment (valentinMachado) [reply packages/core/src/Game/World.js:155]: worldcontext should do operation on go.controller => same structure as browsercontext
worldstatecomputer could a component process ticker something like that

**Comment by valentinMachado** [comment]
context should init controller (actually in assetsManager) scripts should not be in assetsmanager since they dont belong to assets folder anymore

File: packages/core/src/Game/World.js (hunk 4 above)
Comment (valentinMachado) [reply packages/core/src/Game/World.js:28]: origin should then maybe be remove and be declare in config.gameView ?
maybe take the extent and place the world ref at the center ?
world should ref their config path in order to point their geo loc ?

**Comment by valentinMachado** [comment]
hasNoLocalUpdate in go should be rename => isWorldIndependant ou !syncWithWorld ...

**Comment by valentinMachado** [comment]
essayer de se brancher sur la mainloop itowns => remove three js world only from gameview ?

File: packages/browser/src/Widgets/SlideShow/SlideShow.js [hunk 11]
```diff
@@ -192,7 +192,7 @@ export class SlideShow extends Window {
   /** Create all HTMLElements and fill this.htmlSlideShow*/
   initHtml() {
     const htmlSlideShow = document.createElement('div');
-    const coordinatesElement = this.createInputVector(
+    const coordinatesElement = this.fullputVector(
```
Comment (valentinMachado) [inline packages/browser/src/Widgets/SlideShow/SlideShow.js:195]: oups

File: packages/browser/src/Widgets/SlideShow/SlideShow.js [hunk 12]
```diff
@@ -201,7 +201,7 @@ export class SlideShow extends Window {
     this.coordinatesInputVectorID = coordinatesElement.inputVector.id;
     htmlSlideShow.appendChild(coordinatesElement.inputVector);
 
-    const rotationElement = this.createInputVector(
+    const rotationElement = this.fullputVector(
```
Comment (valentinMachado) [inline packages/browser/src/Widgets/SlideShow/SlideShow.js:204]: oups

File: packages/browser/src/Widgets/SlideShow/SlideShow.js [hunk 13]
```diff
@@ -210,7 +210,7 @@ export class SlideShow extends Window {
     this.rotationInputVectorID = rotationElement.inputVector.id;
     htmlSlideShow.appendChild(rotationElement.inputVector);
 
-    const sizeElement = this.createInputVector(
+    const sizeElement = this.fullputVector(
```
Comment (valentinMachado) [inline packages/browser/src/Widgets/SlideShow/SlideShow.js:213]: oups

File: packages/browser/src/Widgets/SlideShow/SlideShow.js [hunk 14]
```diff
@@ -438,7 +438,7 @@ export class SlideShow extends Window {
    * @param {number} step The step of HTMLElement input (type number)
    * @returns {object} title => HTMLElement 'h3' ; inputVector => HTMLElement 'div' contains labels and inputs HTMLElements
    */
-  createInputVector(labels, vectorName, step = 0.5) {
+  fullputVector(labels, vectorName, step = 0.5) {
```
Comment (valentinMachado) [inline packages/browser/src/Widgets/SlideShow/SlideShow.js:441]: oups

**Comment by valentinMachado** [comment]
https://github.com/VCityTeam/UD-Viz/blob/35df3f70022ce337861c894fda5e0b85d396cb58/packages/core/src/Game/State/Interpolator.js#L158

consumed attribute has to be clone in state method

**Comment by valentinMachado** [comment]
https://github.com/VCityTeam/UD-Viz/blob/ecddb182ee5b6c118cc6c6115bf264ae4095233d/packages/node/src/Test.js#L114

should wait a bit after goto to succeed test

**Review by mathieuLivebardon — APPROVED** [review]
🐫 X 1044

**Comment by valentinMachado** [comment]
> camel X 1044

indeed

**Pull request merged**